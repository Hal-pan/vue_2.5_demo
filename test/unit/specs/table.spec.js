import Tables from '@/components/table';
import { shallow } from 'vue-test-utils';

describe('Table', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(Tables);
  });

  it('props', () => {
    const data = wrapper.vm.$options.props.data;
    expect(data.type).toBe(Array);
    expect(wrapper.vm.data).toEqual([]);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('li标签的for循环', () => {
    wrapper.setProps({ data: [1, 2] });
    expect(wrapper.vm.data).toEqual([1, 2]);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('computed test', () => {
    const callback = jest.fn();
    wrapper.vm.$on(wrapper.vm.$options.model.event, callback);
    wrapper.vm.meizhi = 'test';
    expect(callback).toBeCalledWith('test');
    expect(wrapper.vm.test).toBe('test');
  });
});
