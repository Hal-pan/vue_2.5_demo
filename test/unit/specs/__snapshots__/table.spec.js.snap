// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table li标签的for循环 1`] = `
<div>
  <ul>
    <li>
      1
    </li>
    <li>
      2
    </li>
  </ul>
   
  <input
    type="text"
  />
</div>
`;

exports[`Table props 1`] = `
<div>
  <ul />
   
  <input
    type="text"
  />
</div>
`;
